### DISCUSSION TITLE
Custom claims: app_metadata or new key?

### DISCUSSION THREAD

**Discussion by mrjogo** [supabase/supabase#30381] [Questions] [answered]
I'm following (and modifying) https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac#create-auth-hook-to-apply-user-role, which adds a new claim key `user_role` directly to `claims`.

However, on https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook, it says "Claims returned must conform to our specification" and then lists a number of required and optional claims, and the example shows adding `is_admin` into `claims.app_metadata`.

Which is preferred and/or correct?

**Comment by GaryAustin1** [comment, marked as answer]
If I were going to use the auth hooks I would make my own claim.  App metadata gets used for several things and is also in auth.users and the user object table.  The app metadata in auth.users will NOT get changed with this method and just seems like that would be confusing.

**Reply by mrjogo** [reply]
Sounds good! A clarification in the docs of when to use `app_metadata` could be helpful (maybe on https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook)
